device_calendar/ios/Classes/DeviceCalendarPlugin.m

```diff
@@ -1,8 +1,704 @@
+    eventStore = [[EKEventStore alloc] init];
+    validFrequencyTypes = [NSMutableArray new];
+    [validFrequencyTypes addObject: [[NSNumber alloc] initWithInt:EKRecurrenceFrequencyDaily]];
+    [validFrequencyTypes addObject: [[NSNumber alloc] initWithInt:EKRecurrenceFrequencyWeekly]];
+    [validFrequencyTypes addObject: [[NSNumber alloc] initWithInt:EKRecurrenceFrequencyMonthly]];
+    [validFrequencyTypes addObject: [[NSNumber alloc] initWithInt:EKRecurrenceFrequencyYearly]];
+    DeviceCalendarPlugin* instance = [[DeviceCalendarPlugin alloc] init];
+    [registrar addMethodCallDelegate:instance channel:channel];
+}
+
+- (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
+    NSString *method = call.method;
+    if ([method isEqualToString: requestPermissionsMethod]) {
+        [self requestPermissions:nil result:result];
+        return;
+    }
+    else if([method isEqualToString: hasPermissionsMethod]) {
+        [self hasPermissions:nil result:result];
+        return;
+    }
+    else if ([method isEqualToString: retrieveCalendarsMethod]) {
+        [self retrieveCalendars:nil result:result];
+        return;
+    }
+    else if ([method isEqualToString: retrieveEventsMethod]) {
+        [self retrieveEvents:call result:result];
+        return;
+    }
+    else if ([method isEqualToString: createOrUpdateEventMethod]) {
+        [self createOrUpdateEvent:call result:result];
+        return;
+    }
+    else if ([method isEqualToString: deleteEventMethod]) {
+        [self deleteEvent:call result:result];
+        return;
+    }
+    else if ([method isEqualToString:deleteEventInstanceMethod]) {
+        [self deleteEvent:call result:result];
+        return;
+    }
+    else if ([method isEqualToString:createCalendarMethod]) {
+        [self createCalendar:call result:result];
+        return;
+    }
+    else
+        result(FlutterMethodNotImplemented);
+}
+
+-(void)hasPermissions:(id)args result:(FlutterResult)result {
+    result([NSNumber numberWithBool:[self hasEventPermissions]]);
+}
+
+-(void)createCalendar:(FlutterMethodCall *)call result:(FlutterResult)result {
+    NSDictionary *arguments = call.arguments;
+    EKCalendar *calendar = [EKCalendar calendarForEntityType:EKEntityTypeEvent eventStore:eventStore];
+    calendar.title = [arguments valueForKey:calendarNameArgument];
+    NSString *calendarColor = [arguments valueForKey:calendarColorArgument];
+    NSMutableArray *localSources = [NSMutableArray new];
+    NSError *error = nil;
+    if (calendarColor != nil) {
+        calendar.CGColor = [[self colorFromHexString:calendarColor] CGColor];
+    }else {
+        calendar.CGColor = [[[UIColor alloc] initWithRed:255 green:0 blue:0 alpha:0] CGColor];
+    }
+    for (EKSource *ekSource in eventStore.sources) {
+        if (ekSource.sourceType == EKSourceTypeLocal) {
+            [localSources addObject:ekSource];
+        }
+    }
+    if ([localSources count] == 0) {
```

Comment:
When creating a calendar, the code doesn't return any local calendars when I can in Swift - always returning the "Local calendar was not found" error